src/PerformanceSensitiveAnalyzers/CSharp/CodeFixes/AvoidAllocationWithArrayEmptyCodeFix.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            var noAllocation = SyntaxFactory.ParseExpression($"Array.Empty<{typeArgument}>()");
+            var newNode = ReplaceExpression(node, noAllocation);
+            if (newNode == null)
+            {
+                return contextDocument;
+            }
+            var syntaxRootAsync = await contextDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var newSyntaxRoot = syntaxRootAsync.ReplaceNode(node.Parent, newNode);
+            return contextDocument.WithSyntaxRoot(newSyntaxRoot);
+        }
+
+        private static SyntaxNode? ReplaceExpression(SyntaxNode node, ExpressionSyntax newExpression)
+        {
+            switch (node.Parent)
+            {
+                case ReturnStatementSyntax parentReturn:
+                    return parentReturn.WithExpression(newExpression);
+                case ArrowExpressionClauseSyntax arrowStatement:
+                    return arrowStatement.WithExpression(newExpression);
+                case ArgumentListSyntax argumentList:
+                    var newArguments = argumentList.Arguments.Select(x => x == node ? SyntaxFactory.Argument(newExpression) : x);
+                    return argumentList.WithArguments(SyntaxFactory.SeparatedList(newArguments));
+                default:
+                    return null;
+            }
+        }
+
+        private static bool IsCopyConstructor(SemanticModel semanticModel, ObjectCreationExpressionSyntax objectCreation)
+        {
+            if (objectCreation.ArgumentList == null || objectCreation.ArgumentList.Arguments.Count == 0)
+            {
+                return false;
+            }
+
+            if (semanticModel.GetSymbolInfo(objectCreation).Symbol is IMethodSymbol methodSymbol)
+            {
+                if (methodSymbol.Parameters.Any(x => x.Type is INamedTypeSymbol namedType && IsCollectionType(namedType)))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private static bool IsInitializationBlockEmpty(InitializerExpressionSyntax initializer)
+        {
+            return initializer == null || initializer.Expressions.Count == 0;
+        }
+
+        private static bool IsCollectionType(SemanticModel semanticModel, ObjectCreationExpressionSyntax objectCreationExpressionSyntax)
+        {
+            return semanticModel.GetTypeInfo(objectCreationExpressionSyntax).Type is INamedTypeSymbol createdType &&
+                   (createdType.TypeKind == TypeKind.Array || IsCollectionType(createdType));
+        }
+
+        private static bool IsCollectionType(INamedTypeSymbol typeSymbol)
+        {
+            return typeSymbol.ConstructedFrom.Interfaces.Any(x =>
+                x.IsGenericType && x.ToString().StartsWith("System.Collections.Generic.ICollection", System.StringComparison.Ordinal));
```

Comment:
I thought that was better to avoid string based checks for the types and instead rely on type symbol equality?